api/auth.py

```
@@ -95,6 +96,53 @@ def generate_api_key() -> tuple[str, str]:
     hashed_key = hashlib.sha256(raw_key.encode()).hexdigest()
     return raw_key, hashed_key
 
+# Unified authentication dependency that handles both Auth0 and regular JWT tokens
+async def get_current_user_unified(
+    credentials: HTTPAuthorizationCredentials = Depends(security),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)